var/spack/repos/builtin/packages/python/package.py

```diff
@@ -196,6 +196,114 @@ class Python(AutotoolsPackage):
     # An in-source build with --enable-optimizations fails for python@3.X
     build_directory = 'spack-build'
 
+    executables = ['python']
+
+    @classmethod
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        """Allow ``spack external find python`` to locate Python installations.
+
+        Parameters:
+            prefix (str): the directory containing the executables
+            exes_in_prefix (set): the executables that match the regex
+
+        Returns:
+            spack.spec.Spec: the spec of this Python installation
+        """
+        # Possible executable names:
+        # * python, python3, python3.7, python3.7m
+        # Possible red herrings:
+        # * python-config, python3-config, python3.7-config
+        # * spack-python
+        # Take the shortest executable name and hope for the best
+        python = Executable(min(exes_in_prefix))
+        if python.name.startswith('spack-') or python.name.endswith('-config'):
+            return None
```

Comment:
This seems to be working pretty well, and should be flexible enough that we can add new red herrings later